### DISCUSSION TITLE
SSL Certificate Issue

### DISCUSSION THREAD

**Discussion by satya-devloper** [micronaut-projects/micronaut-core#8616] [Q&A] [answered]
I have two micronaut services, and I m using self signed true, in order to communicate both services. When I was using micronaut 2.3.3 , everything was fine. But switching to 3.7.5 is giving certificate_unknow exception

PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

In first micronaut service my config looks like
micronaut:
  application:
    name: m1
  server:
    port: 8443
    max-request-size: '500MB'
    multipart:
      max-file-size: '350MB'
    cors:
      enabled: true
  ssl:
    enabled: true
    buildSelfSigned: true
    
    
In second micronaut service my config looks like
micronaut:
  application:
    name: m2
  server:
    port: 8443
  ssl:
    enabled: true
    buildSelfSigned: true

Please let me know what extra thing I m missing

**Comment by yawkat** [comment, marked as answer]
Are you using the Micronaut HTTP client? You may have to set the `micronaut.http.client.ssl.insecure-trust-all-certificates` property. But be aware, this basically defeats the point of using SSL in the first place, so you may want to just switch to HTTP.

**Reply by satya-devloper** [reply]
Since I m using Micronaut 3 , So I m using rxjava2-http-client as per the documentation suggestion. So you mean If I use micronaut.http.client.ssl.insecure-trust-all-certificates  then although it runs on https , cert will be ignored and connection should be made ?? (I tried this also but no luck)
  
I want to know what changes has been done since 2.3.3 to 3.7.5, that it stopped working. Was earlier ssl verification ignored inspite of using ssl?

Also suppose I want to use localhost cert, how can I implement to interact two micronaut services.

**Reply by yawkat** [reply]
yes it was ignored previously. it was fixed around 3.1 i think.

you can also create your own key store and add it to both the client and server, as shown here: https://docs.micronaut.io/latest/guide/index.html#https

**Reply by satya-devloper** [reply]
In my second micronaut service I configured like this:
micronaut:
  http:
    client:
      ssl:
        insecureTrustAllCertificates: true
  application:
    name: m2
  server:
    port: 8081
  ssl:
    enabled: true
  executors:
    io:
      type: fixed
      nThreads: 10



but when first micronuat service Is calling it via rxjava2-http-client, it is giving handshake error. Please let me know, am I missing any thing

**Reply by yawkat** [reply]
you need to set the property on the client, not on the server

**Reply by satya-devloper** [reply]
Thank you so much , its solved with both cert and without cert.It was my mistake , I was adding at server side
